lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -279,33 +282,132 @@ export abstract class ClientApplication {
             validRequest.correlationId,
             validRequest.forceRefresh
         );
+
         try {
-            const silentFlowClientConfig =
+            const clientConfiguration =
                 await this.buildOauthClientConfiguration(
                     validRequest.authority,
                     validRequest.correlationId,
                     validRequest.redirectUri || "",
                     serverTelemetryManager,
                     undefined,
-                    request.azureCloudOptions
+                    validRequest.azureCloudOptions
                 );
-            const silentFlowClient = new SilentFlowClient(
-                silentFlowClientConfig
-            );
+            const silentFlowClient = new SilentFlowClient(clientConfiguration);
             this.logger.verbose(
                 "Silent flow client created",
                 validRequest.correlationId
             );
-            return await silentFlowClient.acquireToken(validRequest);
+
+            try {
+                return await this.acquireCachedToken(
+                    validRequest,
+                    silentFlowClient,
+                    clientConfiguration
+                );
+            } catch (e) {
+                try {
+                    // If persistence exists, overwrite the in-memory cache with the persistent cache and search for the cached token again
+                    const persistence = this.config.cache.cachePlugin;
+                    if (persistence) {
+                        this.logger.info(
+                            "Overwriting in-memory cache with persistent cache"
+                        );
+                        this.storage.clear();
+                        const cacheContext = new TokenCacheContext(
+                            this.tokenCache,
+                            false
+                        );
+                        await persistence.beforeCacheAccess(cacheContext);
+                        const cacheSnapshot =
+                            this.tokenCache.getCacheSnapshot();
+                        this.storage.setCache(cacheSnapshot);
+                        await persistence.afterCacheAccess(cacheContext);
```

Comment:
Piggy-backing off the previous comment, could/should this be added as a new function in `TokenCache.ts`?